src/workerd/server/tests/compile-tests/compile-test.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# $1 -> workerd binary path
+# $2 -> path to desired file to compile
+# $3 -> port to curl
+# $4 -> desired output
+
+# Help Function
+function show_help() {
```

Comment:
I would convert show_help to an elaborate file-based comment where you document besides the command line the purpose of the test.

Reply:
I added more to it, however I think the ability to use shell scripts like shell scripts is helpful for development.